### ISSUE TITLE
[slim] feature encode / decode a transaction

### ISSUE THREAD

**Issue by jcalfee** [EOSIO/eos#3010]
Can we have functions like `sign` that will `encode` and `decode` a transaction between JSON and HEX?
 
```bash
cleos transaction sign '{"expiration":"2018-05-13T15:22:27","ref_block_num":6407,"ref_block_prefix":2487141821,"net_usage_words":0,"max_cpu_usage_ms":0,"delay_sec":0,"context_free_actions":[],"actions":[{"account":"eosio.token","name":"transfer","authorization":[{"actor":"inita","permission":"active"}],"data":"000000000093dd74000000008093dd74102700000000000004454f530000000000"}],"transaction_extensions":[]}'
```

This is going to help other libraries debug serialization and signing ..

**Comment by tbfleming** [comment]
Might want to support action's data in json form also. That'd completely relieve other libraries from having to serialize.

**Comment by tbfleming** [comment]
Here's a potential attack that users would need to be aware of:
* Library uses the new API to encode a transaction.
* API runs on a malicious server and encodes something different. e.g. send all EOS to attacker's account.
* Library signs the encoded transaction. Since it doesn't know how to deserialize itself, it doesn't recognize the tampering.

**Comment by jcalfee** [comment]
Well, if you want to avoid that attack it would be nice to have that code organized so a specific wasm library can be created of reasonable size.  That library can structure by name and with JSON or binary and and encode / decode it.  That will make it more portable and language independent.

If i is a reasonable size will make libraries like eosjs very easy to create and maintain..

**Comment by jcalfee** [comment]
We worked out the error another way so this is a low priority.. The error was related to a change in the transaction signing digest not the actual transaction itself and not related to transaction encoding.  

As a work-around for the above, I use the cleos wallet to create a similar transaction then use `cleos get transaction <id>` to view the hex_data for a given action.  That does not give me the full transaction hex, but it it close.  In the long-run would prefer to try and wasm compile FC though so we only have to be concerned with serialization in one place..

**Comment by wanderingbort** [comment]
this seems implemented by #5225

**Comment by BigMurry** [comment]
I made a little tool to decode packed eos tx.

https://github.com/BigMurry/eos-unpack
